src/sst/core/interfaces/stdMem.h

```diff
@@ -0,0 +1,748 @@
+        virtual ~MoveData() {}
+
+        virtual Request* makeResponse() override { return new WriteResp(id, pDst, size, flags, vDst, iPtr, tid); }
+        virtual bool needsResponse() override { return posted; }
+        SST::Event* convert(RequestConverter* converter) override { return converter->convert(this); }
+        void handle(RequestHandler* handler) override { return handler->handle(this); }
+
+        /* Data members */
+        Addr pSrc;   /* Physical address of source */
+        Addr vSrc;  /* Virtual address of source */
+        Addr pDst;  /* Physical address of destination */
+        Addr vDst;  /* Virtual address of destination */
+        uint64_t size;  /* Number of bytes to move */
+        bool posted;    /* True if a response is needed */
+        Addr iPtr;      /* Instruction pointer */
+        uint32_t tid;   /* Thread ID */
+    };
+
+    /** Notifies endpoint that an address has been invalidated from the L1. 
+     */
+    class InvNotify : public Request {
+    public:
+        InvNotify(Addr pAddr, uint64_t size, flags_t flags = 0, Addr vAddr = 0, Addr iPtr = 0, uint32_t tid = 0) :
+            Request(flags), pAddr(pAddr), vAddr(vAddr), size(size), iPtr(iPtr), tid(tid) {}
+        virtual ~InvNotify() {}
+
+        virtual Request* makeResponse() override { return nullptr; }
+        virtual bool needsResponse() override { return false; }
+        SST::Event* convert(RequestConverter* converter) override { return converter->convert(this); }
+        void handle(RequestHandler* handler) override { return handler->handle(this); }
+
+        Addr pAddr; /* Physical address */
+        Addr vAddr; /* Virtual address */
+        uint64_t size;  /* Number of bytes invalidated */
+        Addr iPtr;      /* Instruction pointer */
+        uint32_t tid;   /* Thread ID */
+    };
+
+    /* This class can be inherited to create custom events that can be handled in a limited fashion by existing interfaces
+     * Child class must be serializable 
+     */
+    class CustomData : public SST::Core::Serialization::serializable {
+    public:
+        CustomData() {}
+        virtual ~CustomData() {}
+        virtual Addr getRoutingAddress() = 0;   /* Return address to use for routing this event to its destination */
+        virtual uint64_t getSize() = 0;         /* Return size to use when accounting for bandwidth used is needed */
+        virtual CustomData* makeResponse() = 0; /* Return a CustomData* object formatted as a response */
+        virtual bool needsResponse() = 0;       /* Return whether a response is needed */
+
+        /* This needs to be serializable so that we can use it in events in parallel simulations */
+        virtual void serialize_order(SST::Core::Serialization::serializer &UNUSED(ser)) override = 0;
+        //ImplementSerializable(SST::Experimental::Interfaces::StandardMem::CustomData);
```

Comment:
You could use the ImplementVirtualSerializable macro.  I don't think it's strictly needed, but it does tag the class appropriately and would be forward compatible if the macro becomes required in the future.  Also, it would need to be called with just the class name, not the fully namespaced name.